codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerEnumGeneratorTest.kt

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy.generators
+
+import io.kotest.matchers.string.shouldContain
+import org.junit.jupiter.api.Test
+import software.amazon.smithy.model.shapes.StringShape
+import software.amazon.smithy.model.traits.EnumTrait
+import software.amazon.smithy.rust.codegen.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.smithy.CodegenMode
+import software.amazon.smithy.rust.codegen.testutil.TestRuntimeConfig
+import software.amazon.smithy.rust.codegen.testutil.asSmithyModel
+import software.amazon.smithy.rust.codegen.testutil.compileAndTest
+import software.amazon.smithy.rust.codegen.testutil.testSymbolProvider
+import software.amazon.smithy.rust.codegen.util.expectTrait
+import software.amazon.smithy.rust.codegen.util.lookup
+
+class ServerEnumGeneratorTest {
+    @Test
+    fun `it generates named enums`() {
+        val model = """
+            namespace test
+            @enum([
+                {
+                    value: "t2.nano",
+                    name: "T2_NANO",
+                    documentation: "T2 instances are Burstable Performance Instances.",
+                    tags: ["ebsOnly"]
+                },
+                {
+                    value: "t2.micro",
+                    name: "T2_MICRO",
+                    documentation: "T2 instances are Burstable Performance Instances.",
+                    tags: ["ebsOnly"]
+                },
+            ])
+            string InstanceType
+        """.asSmithyModel()
+        val provider = testSymbolProvider(model)
+        val writer = RustWriter.forModule("model")
+        val shape = model.lookup<StringShape>("test#InstanceType")
+        val generator = ServerEnumGenerator(model, provider, writer, shape, shape.expectTrait(), CodegenMode.Server, TestRuntimeConfig)
+        generator.render()
+        writer.compileAndTest(
+            """
+            let instance = InstanceType::T2Micro;
+            assert_eq!(instance.as_str(), "t2.micro");
+            assert_eq!(InstanceType::try_from("t2.nano").unwrap(), InstanceType::T2Nano);
+            // check no unknown
+            match instance {
+            InstanceType::T2Micro => (),
+            InstanceType::T2Nano => (),
+            }
+            """
+        )
+
+        writer.toString() shouldContain "#[non_exhaustive]"
```

Comment:
Now that the `Unknown` variant is no longer there, we also don't need to generate `#[non_exhaustive]` on server enums. The server always has the latest version of the model; this will allow users to exhaustively `match` on their enums.